### PR TITLE
[Feature] Make only certain layers editable / draggable / removeable / cutable

### PR DESCRIPTION
**Pull request by Falke-Design**
New options:
- allowEditing
- allowDragging
- allowRemoval
- allowCutting

Also there is a new option `layersToCut: [layer1, layer2, ...]` for the Cutting Mode, to cut only the layers in the array.

Fixes: #583, #276, #250, #420

### CODE REVIEW

File: cypress/integration/polygon.spec.js [hunk 1]
```diff
@@ -932,4 +932,123 @@ describe('Draw & Edit Poly', () => {
...
+
+    cy.toolbarButton('edit').click();
+    cy.get(mapSelector)
+      .rightclick(160, 50);
+
+    cy.window().then(({ map }) => {
+      expect(1).to.eq(map.pm.getGeomanDrawLayers().length);
+    });
+  });
+
```
Comment (Falke-Design) [inline cypress/integration/polygon.spec.js:955]: Test if only one of two Layers are disabled for editing

File: src/js/Draw/L.PM.Draw.Cut.js [hunk 2]
```diff
@@ -84,6 +84,18 @@ Draw.Cut = Draw.Polygon.extend({
       .filter(l => l instanceof L.Polyline)
       // exclude the drawn one
       .filter(l => l !== layer)
+      // layer is allowed to cut
+      .filter(l => l.pm.options.allowCutting)
+      // filter out everything that ignore leaflet-geoman
+      .filter(l =>{
+        // TODO: after cutting nothing else can be cutted anymore until a new list is passed, because the layers don't exists anymore. Should we remove the cutted layers from the list?
+        if(this.options.layersToCut && L.Util.isArray(this.options.layersToCut) && this.options.layersToCut.length > 0){
```
Comment (Falke-Design) [inline src/js/Draw/L.PM.Draw.Cut.js:92]: Cutted layers are removed from the list -> Add also to readme

File: src/js/Edit/L.PM.Edit.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ import RotateMixin from "../Mixins/Rotating";
 const Edit = L.Class.extend({
   includes: [DragMixin, SnapMixin, RotateMixin],
   options: {
-    snappable: true,
+    snappable: true, //TODO: should we create also an option allowSnapping
```
Comment (Falke-Design) [inline src/js/Edit/L.PM.Edit.js:8]: use allowSnapping and make snappable deprecated

File: src/js/Edit/L.PM.Edit.js [hunk 4]
```diff
@@ -14,6 +14,10 @@ const Edit = L.Class.extend({
     limitMarkersToCount: -1,
     hideMiddleMarkers: false,
     draggable: true,
+    allowEditing: true, //TODO: should it disable all interactions on a layer if `enable()` is called? For example a Circle can also be dragged -> But it is in Edit-Mode
```
Comment (Falke-Design) [inline src/js/Edit/L.PM.Edit.js:17]: Yes should also disabled

File: src/js/Edit/L.PM.Edit.js [hunk 5]
```diff
@@ -14,6 +14,10 @@ const Edit = L.Class.extend({
...
+    allowDragging: true, //Todo: same as draggable? -> Not direct but can confuse the user. Draggable are only options for Marker to make it draggable (Leaflet Core)
```
Comment (Falke-Design) [inline src/js/Edit/L.PM.Edit.js:18]: use allowDragging and make draggable deprecated -> draggable of Leaflet will be overwritten

File: src/js/Edit/L.PM.Edit.js [hunk 6]
```diff
@@ -14,6 +14,10 @@ const Edit = L.Class.extend({
...
+    allowRemoval: true, //TODO: or should we name it "allowDeleting"
```
Comment (Falke-Design) [inline src/js/Edit/L.PM.Edit.js:19]: we use allowRemoval

File: src/js/Edit/L.PM.Edit.js (hunk 3 above)
Comment (Falke-Design) [reply src/js/Edit/L.PM.Edit.js:8]: I suggest to do this in the next major release

File: src/js/Edit/L.PM.Edit.js (hunk 5 above)
Comment (Falke-Design) [reply src/js/Edit/L.PM.Edit.js:18]: I suggest to do this in the next major release

**Review by codeofsumit — APPROVED** [review]

**Pull request merged**